compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp

```diff
@@ -0,0 +1,210 @@
+    {
+      bias->at<loco::DataType::FLOAT32>(i) *= multiplication->at<loco::DataType::FLOAT32>(i);
+    }
+  }
+}
+
+/**
+ *  Fuse Mul to FullyConnected if the multiplied value is a channel(last dimension)-wise constant
+ *
+ *  BEFORE
+ *                |
+ *      [CircleFullyConnected]
+ *                |
+ *           [CircleMul]
+ *                |
+ *
+ *  AFTER
+ *                |
+ *       [CircleFullyConnected]   [CircleMul] (dead)
+ *                |
+ *
+ */
+bool fuse_mul_with_fc(luci::CircleFullyConnected *fc)
+{
+  // Sanity check:
+  RETURN_FALSE_UNLESS(fc);
+  // Allow only FLOAT32 data type:
+  RETURN_FALSE_UNLESS(fc->dtype() == loco::DataType::FLOAT32);
+  // Allow only without activation functions as values are going to
+  // be multiplied before activation function.
+  RETURN_FALSE_UNLESS(fc->fusedActivationFunction() == luci::FusedActFunc::NONE);
+  // Check for weights being Constant:
+  auto weights = dynamic_cast<luci::CircleConst *>(fc->weights());
+  RETURN_FALSE_UNLESS(weights);
+  // Get Mul node:
+  auto fc_output = loco::succs(fc);
+  // Make sure that FullyConnected has only one child:
+  RETURN_FALSE_UNLESS(fc_output.size() == 1);
+  auto mul = dynamic_cast<luci::CircleMul *>(*fc_output.begin());
+  RETURN_FALSE_UNLESS(mul);
+  // Allow Mul node only with FLOAT32 data type:
+  RETURN_FALSE_UNLESS(mul->dtype() == loco::DataType::FLOAT32);
+  // Get multiplication Constant (here: the second input besides weights):
+  auto multiplication = mul->x() == fc ? dynamic_cast<luci::CircleConst *>(mul->y())
+                                       : dynamic_cast<luci::CircleConst *>(mul->x());
+  RETURN_FALSE_UNLESS(multiplication);
+  // Get rank of multiplication:
+  auto rank = multiplication->rank();
+  RETURN_FALSE_UNLESS(rank != 0);
+  // Check that all dimensions are ones, checks broadcast capabilites.
+  // Last dimesion of multiplication must be compatible with FC.
+  // N-D case (N>1):
+  if (multiplication->rank() > 1)
+  {
+    // Check channel-wise broadcasting:
+    for (uint32_t i = 0; i < rank - 1; i++)
+      RETURN_FALSE_UNLESS(multiplication->dim(i).value() == 1);
+    // Check the last dimesion of Mul is the same with the first dimension of FullyConnected
+    RETURN_FALSE_UNLESS(multiplication->dim(rank - 1) == weights->dim(0));
+  }
+  // Scalar case:
+  else if (multiplication->rank() == 1 || multiplication->rank() == 0)
+  {
+    RETURN_FALSE_UNLESS(multiplication->size<loco::DataType::FLOAT32>() != 0);
+  }
+
+  // Only supports:
+  // (1) constant bias
+  // (2) no bias
+  auto bias = loco::must_cast<luci::CircleNode *>(fc->bias());
+  RETURN_FALSE_UNLESS(bias->opcode() == luci::CircleOpcode::CIRCLECONST or
+                      bias->opcode() == luci::CircleOpcode::CIRCLEOUTPUTEXCLUDE)
+  // Create new bias to be updated with values:
+  auto const_bias = dynamic_cast<luci::CircleConst *>(fc->bias());
+  RETURN_FALSE_UNLESS(const_bias)
```

Comment:
This condition prevents support for `no bias`, which contradicts with the above comment.

Reply:
Could you add a test for `no bias`?

Reply:
Great catch, I hope that I was able to handle it according to project standards. Test is also added.

Should I create additional tflite recipe for it?

Reply:
Yes please.

Reply:
Here is the PR: https://github.com/Samsung/ONE/pull/13650